cli/ceph/orch/orch.py

```diff
@@ -0,0 +1,132 @@
+                                  the orchestrator(Optional).
+                    format(str): the type to be formatted(yaml)(Optional).
+                    refresh(bool): Whether to refresh or not(Optional).
+        """
+        args = ""
+        for k, v in kw.items():
+            if v is True:
+                args = args + f" --{k}"
+            else:
+                args = args + f" --{k} {v}"
+        cmd = f"{self.base_cmd} ls {args}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def apply(self, **kw):
+        """
+        Applies the configuration to hosts.
+        Args:
+          cluster : Ceph cluster
+          kw (Dict): Key/value pairs that needs to be provided to the installer.
+          Example:
+            Supported keys:
+              service-name(str): name of service.
+              hosts(list): name of host(Optional).
+              placement(dict): Placement on hosts(Optional).
+              count(int): No of hosts to be applied(Optional)
+        Returns:
+          Dict(str):  A mapping of host strings of the given task's return
+                      value for that host's execution run.
+        """
+        service_name = kw.pop("service", "")
+        cmd = f"{self.base_cmd} apply {service_name}"
+
+        pos_args = kw.get("pos_args")
+        if pos_args:
+            for arg in pos_args:
+                cmd += f" {arg}"
+
+        args = kw.get("args")
+
+        placement = args.pop("placement", {})
+
+        if placement:
+            placement_str = "--placement="
+            sep = placement.get("sep", " ")
+            node_names = list()
+
+            if "label" in placement:
+                node_names.append(f"label:{placement['label']}")
+
+            if "nodes" in placement:
+                nodes = placement.get("nodes")
+
+                if "*" in nodes:
+                    node_names.append("*")
+                elif "[" in nodes:
+                    placement_str += f'"{nodes}"'
+                else:
+                    nodes_ = get_nodes_by_ids(self.ctx, node_names)
+                    node_names = [node.hostname for node in nodes_]
+
+            # Support RGW count-per-host placement option
+            if "count-per-host" in placement and service_name == "rgw":
+                node_names.append(f"count-per-host:{placement['count-per-host']}")
+
+            node_str = f"{sep}".join(node_names)
+
+            limit = placement.pop("limit", None)
+            if limit:
+                placement_str += f"'{limit}{sep}{node_str}'"
+            else:
+                placement_str += f"'{node_str}'"
+
+            cmd += placement_str
+
+        # Add the arguments to the cmd
+        if args:
+            cmd += config_dict_to_string(args)
+
+        out, _ = self.execute(sudo=True, cmd=cmd)
+
+        if not out:
+            raise OrchApplyServiceFailure(service_name)
```

Comment:
This is not required here. When you set `check_ec=False` which is default value here, ceph.exec_command raises exception.